internal/ingress/controller/nginx.go

```diff
@@ -122,6 +122,7 @@ func NewNGINXController(config *Configuration, mc metric.Collector) *NGINXContro
 
 	n.store = store.New(
 		config.Namespace,
+		config.WatchNamespaceSelector,
```

Comment:
What if instead of adding a new argument, we turn config.Namespace into config.WatchNamespaceSelector with only one namespace when --namespace is specified?

Reply:
Wouldn't that force installation with clusterroles rather then single namespaced installs that could do it to a single namespace without cluster wide rules?

Reply:
I think your concern is correct @kfox1111 but looking at the code, it seems that the namespace watcher just tries to work if you add the selector namespace.

Anyway, I have some concerns as well if adding a selector would demand full cluster namespace watch, and want some more opinions on that